### DISCUSSION TITLE
Enable Mermaid rendering in Docusaurus

### DISCUSSION THREAD

**Discussion by ds-lcapellino** [catenax-eV/catenax-ev.github.io#114] [Ideas]
Hey everyone, 

I think it would be great to support Mermaid rendering in Docusaurus to easily create Diagram and display them on https://catenax-ev.github.io/docs/next/working-model/infrastructure. 
As you can see, the diagram there currently does not render. It would be easier to render diagrams in comparison to updating screenshots. Also look and feel would be better. 

Docusaurus already provides this feature https://docusaurus.io/docs/next/markdown-features/diagrams. We just need to implement it. 
What do you guys think?

**Comment by stephanbcbauer** [comment]
@jSchuetz88

**Reply by stephanbcbauer** [reply]
Would like to activate both, PLANT UML and mermaid, if possible

**Comment by ds-lcapellino** [comment]
@stephanbcbauer, @jSchuetz88 
I've created PR https://github.com/catenax-eV/catenax-ev.github.io/pull/115 to introduce Mermaid rendering. Reading the docs, I'll guess that Mermaid is supported out of the box, while PlantUML requires additional Plugins.

**Reply by jSchuetz88** [reply]
I just checked, you are right. Enabling Mermaid won't be an issue. But PlantUML we have to check, because it is mandatory that the images are rendered locally. But it should also be possible with `remark-local-plantuml `.
I will check and test both in the course of this month and provide an update.

From perspective we have 3 pre-conditions for the usage:

- Github needs to be able to render them natively, as the standards are not maintaned with / in docusaurus.
- The images need to be rendered either locally or at our own environment
- They need to be integretable with docusaurus without breaking its maintainability

**Reply by ds-lcapellino** [reply]
Regarding Mermaid, GitHub is able to render it out of the box, see https://github.com/catenax-eV/cx-working-model/blob/main/docs/infrastructure/infrastructure.md

**Comment by jSchuetz88** [comment]
I think we can close this discussion. Based on this discussion we enabled Mermaid support a while ago. At the current state I don't see any added value to support two different ways to create / embed diagrams (at least for the standards I want to avoid to have two different notations).
